src/diagnosticManager.ts

```diff
@@ -48,6 +54,41 @@ export class DiagnosticManager {
     );
   }
 
+  getMixedErrorsAndWarns(root: string): [Set<string>, Set<string>] {
+    const errorPaths: Set<string> = new Set();
+    const warningPaths: Set<string> = new Set();
+
+    // Get the errors
+    Object.entries(this.errorPathCount).forEach(([fullpath, _]) => {
+      const relativePath = pathLib.relative(root, fullpath);
+      const parts = relativePath.split(pathLib.sep);
+
+      for (let i = 1; i <= parts.length; i++) {
+        const frontalPath = pathLib.join(
+          root,
+          parts.slice(0, i).join(pathLib.sep),
+        );
+        errorPaths.add(frontalPath);
+      }
+    });
+
+    // Get the warnings
+    Object.entries(this.warningPathCount).forEach(([fullpath, _]) => {
+      const relativePath = pathLib.relative(root, fullpath);
+      const parts = relativePath.split(pathLib.sep);
+
+      for (let i = 1; i <= parts.length; i++) {
+        const frontalPath = pathLib.join(
+          root,
+          parts.slice(0, i).join(pathLib.sep),
+        );
+        warningPaths.add(frontalPath);
+      }
+    });
```

Comment:
Why not call `getMixedError` and `getMixedWarning` directly, are they different?